src/android/android-uvc-device.cpp

```diff
@@ -0,0 +1,596 @@
+            // chances are that we will need to adjust some of these operation to the
+            // processing unit and some to the camera terminal.
+            unit = _processing_unit;
+            switch (option) {
+                case RS2_OPTION_BACKLIGHT_COMPENSATION:
+                    return UVC_PU_BACKLIGHT_COMPENSATION_CONTROL;
+                case RS2_OPTION_BRIGHTNESS:
+                    return UVC_PU_BRIGHTNESS_CONTROL;
+                case RS2_OPTION_CONTRAST:
+                    return UVC_PU_CONTRAST_CONTROL;
+                case RS2_OPTION_EXPOSURE:
+                    unit = _input_terminal;
+                    return UVC_CT_EXPOSURE_TIME_ABSOLUTE_CONTROL;
+                case RS2_OPTION_GAIN:
+                    return UVC_PU_GAIN_CONTROL;
+                case RS2_OPTION_GAMMA:
+                    return UVC_PU_GAMMA_CONTROL;
+                case RS2_OPTION_HUE:
+                    return UVC_PU_HUE_CONTROL;
+                case RS2_OPTION_SATURATION:
+                    return UVC_PU_SATURATION_CONTROL;
+                case RS2_OPTION_SHARPNESS:
+                    return UVC_PU_SHARPNESS_CONTROL;
+                case RS2_OPTION_WHITE_BALANCE:
+                    return UVC_PU_WHITE_BALANCE_TEMPERATURE_CONTROL;
+                case RS2_OPTION_ENABLE_AUTO_EXPOSURE:
+                    unit = _input_terminal;
+                    return UVC_CT_AE_MODE_CONTROL; // Automatic gain/exposure control
+                case RS2_OPTION_ENABLE_AUTO_WHITE_BALANCE:
+                    return UVC_PU_WHITE_BALANCE_TEMPERATURE_AUTO_CONTROL;
+                case RS2_OPTION_POWER_LINE_FREQUENCY:
+                    return UVC_PU_POWER_LINE_FREQUENCY_CONTROL;
+                case RS2_OPTION_AUTO_EXPOSURE_PRIORITY:
+                    unit = _input_terminal;
+                    return UVC_CT_AE_PRIORITY_CONTROL;
+                default:
+                    throw linux_backend_exception(to_string() << "invalid option : " << option);
+            }
+        }
+
+        // Retreive USB data, sign bit is calculated according to length
+        int32_t android_uvc_device::get_data_usb(uvc_req_code action, int control, int unit,
+                                                 unsigned int length) const {
+            unsigned char buffer[4] = {0};
+            int32_t ret = 0;
+
+            int status = usb_device_control_transfer(
+                    _device->device->get_handle(),
+                    UVC_REQ_TYPE_GET,
+                    action,
+                    control << 8,
+                    unit << 8 | (_device->deviceData.ctrl_if.bInterfaceNumber),
+                    buffer,
+                    sizeof(int32_t), 0);
+
+            if (status < 0) {
+                throw ("usb_device_control_transfer failed!");
+            }
+
+            if (status != sizeof(int32_t)) {
+                throw ("insufficient data read from USB");
```

Comment:
You forgot `std::runtime_error`, same for line 162

Reply:
I've addressed this in our internal fork.